vertical-pod-autoscaler/pkg/updater/priority/priority_processor_test.go

```diff
@@ -0,0 +1,312 @@
+	vpa := test.VerticalPodAutoscaler().WithName("test-vpa").WithContainer("test-container").Get()
+	result := p.GetUpdatePriority(pod, vpa, nil)
+	assert.NotNil(t, result)
+}
+
+func TestGetUpdatePriority_VpaObservedContainers(t *testing.T) {
+	const (
+		// There is no VpaObservedContainers annotation
+		// or the container is listed in the annotation.
+		optedInContainerDiff = 9
+		// There is VpaObservedContainers annotation
+		// and the container is not listed in.
+		optedOutContainerDiff = 0
+	)
+	testVpa := test.VerticalPodAutoscaler().WithName("test-vpa").WithContainer(containerName).Get()
+	tests := []struct {
+		name           string
+		pod            *corev1.Pod
+		recommendation *vpa_types.RecommendedPodResources
+		want           float64
+	}{
+		{
+			name:           "with no VpaObservedContainers annotation",
+			pod:            test.Pod().WithName("POD1").AddContainer(test.BuildTestContainer(containerName, "1", "")).Get(),
+			recommendation: test.Recommendation().WithContainer(containerName).WithTarget("10", "").Get(),
+			want:           optedInContainerDiff,
+		},
+		{
+			name: "with container listed in VpaObservedContainers annotation",
+			pod: test.Pod().WithAnnotations(map[string]string{annotations.VpaObservedContainersLabel: containerName}).
+				WithName("POD1").AddContainer(test.BuildTestContainer(containerName, "1", "")).Get(),
+			recommendation: test.Recommendation().WithContainer(containerName).WithTarget("10", "").Get(),
+			want:           optedInContainerDiff,
+		},
+		{
+			name: "with container not listed in VpaObservedContainers annotation",
+			pod: test.Pod().WithAnnotations(map[string]string{annotations.VpaObservedContainersLabel: ""}).
+				WithName("POD1").AddContainer(test.BuildTestContainer(containerName, "1", "")).Get(),
+			recommendation: test.Recommendation().WithContainer(containerName).WithTarget("10", "").Get(),
+			want:           optedOutContainerDiff,
+		},
+		{
+			name: "with incorrect VpaObservedContainers annotation",
+			pod: test.Pod().WithAnnotations(map[string]string{annotations.VpaObservedContainersLabel: "abcd;';"}).
+				WithName("POD1").AddContainer(test.BuildTestContainer(containerName, "1", "")).Get(),
+			recommendation: test.Recommendation().WithContainer(containerName).WithTarget("10", "").Get(),
+			want:           optedInContainerDiff,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
+			p := NewProcessor()
+			result := p.GetUpdatePriority(tc.pod, testVpa, tc.recommendation)
+			assert.NotNil(t, result)
+			// The resourceDiff should be a difference between container resources
+			// and container resource recommendations. Containers not listed
+			// in an existing vpaObservedContainers annotations shouldn't be taken
+			// into account during calculations.
+			assert.InDelta(t, result.resourceDiff, tc.want, 0.0001)
+		})
+	}
+}
+
+type FakePriorityProcessor struct {
```

Comment:
We don't have to export FakePriorityProcessor. We can return PriorityProcessor interface.

Reply:
Done